File: 409.cpp
// done in cpp longest palindrome leetcode 
#include<bits/stdc++.h>
#include<iostream>
using namespace std;

int longest(string s){
  int length =0;
  set<char> st;
  for(char elem:s){
    if(st.find(elem)!=st.end()){
      st.remove(elem);
      length+=2;
    }
    else{
      st.insert(elem);
    }
  }
  if(!st.empty()){
    length++;
  }
  return length;
}

int main()
{
  return 0;
}

File: 409.py
# problem no 409 leetcode "longest palindrome"
def longest(s):
    charSet = set()
    length =0
    for elem in s:
        if elem in charSet:
            charSet.remove(elem)
            length+=2
        else:
            charSet.add(elem)
    if charSet:
        length+=1
    print(length)


s = "abccccdd"
longest(s)
